doc/doxygen/src/driver-guide.md

```diff
@@ -266,18 +266,28 @@ int16_t tmpabc_read(const tmpabc_t *dev)
 }
 @endcode
 
-Whenever you implement status/error return values in your driver, they
-should be named, meaning that the driver MUST define an enum assigning names to
-the actual used value, e.g.
+Whenever you implement status/error return values in your driver, magic numbers
+must ***not*** be used. Instead, use negative `errno` codes (such as `-EIO` for
+an input/output error) and *document* every single error code that can be
+return for each function using the `reval` Doxygen command. E.g. like this:
```

Comment:
```suggestion
returned for each function using the `retval` Doxygen command. E.g. like this:
```
In general, we (well I) use `@return` instead `@retval`.

Reply:
The difference is that `@return` describes what is returned in general, and `@retval` documents a specific value.

E.g. 

```C
/**
 * @brief Receive a frame
 * @param[out] dest   Destination buffer
 * @param[in]  lenght Lenght of @p dest
 *
 * @return             Number of bytes written
 * @retval -EOVERFLOW  Destionation buffer to small
 * @retval -EIO        I/O error while receiving the frame
 */
```

So using `@return` is correct. Only when you want to document a "special" return code, `@retval` is the command to use. And of course, you can mix them if needed.

And e.g. when having a binary function, e.g. `bool is_char_numeric(char input)`, you could also go for just using `@retval` for each of the two possible return values and provide no `@return` description.

Reply:
Do you think it would make sense to mention that in this document ? I have the impression that there's an abuse of `@return` in the current RIOT documentation (in many places).